netstandard2.0 compatiblity

arendvw opened the pull request:
Thanks for the effort in this library.

This project is currently only compatible with dotnet 6 and above, but the code and dependencies are trivially compatible with .NET Standard 2.0 as well. 

Here is a pull request to target .net standard 2.0, and test against net48, dotnet 6 and dotnet 8.

I encountered a small case of floating point instability, and added additional code to allow checking of equality against double.delta (my guess is that the source data was somehow generated on a different platform, on my windows there seem to be more digits for the triangulation results).

All tests succeed here.

Would you be willing to release this library as .net standard 2.0?

arendvw committed:
refactor:
- Refocus project to target .netstandard 2.0, update tests to run as .net core 6, 8 and .net48.
- Convert sln to slnx

arendvw committed:
tests: updated polyhedron equality tests. Previous tests were not stable due to small floating point errors in text comparison. (Probably because I run these tests on windows and the source tests were on windows?).

bertt commented:
thanks, looks good.

I think we can simplify the polyhedralsurface test by only checking number of vertices (and not the positions) - so all the code with precision is not needed.

```
            var polyhedralsurface = (PolyhedralSurface)Geometry.Deserialize<WktSerializer>(expected);
            var wkb = polyhedralsurface.AsBinary();

            // act
            var polyhedralsurface_after = (PolyhedralSurface)Geometry.Deserialize<WkbSerializer>(wkb);

            Assert.That(polyhedralsurface_after.Geometries.Count, Is.EqualTo(polyhedralsurface.Geometries.Count));

            for (var i = 0; i < polyhedralsurface.Geometries.Count; i++)
            {
                var expectedPolygon = polyhedralsurface.Geometries[i];
                var actualPolygon = polyhedralsurface_after.Geometries[i];

                Assert.That(actualPolygon.ExteriorRing.Points.Count, Is.EqualTo(expectedPolygon.ExteriorRing.Points.Count));
            }
```

arendvw committed:
test: replaced testing to a simoler way of testing if polyhedrons are the same

arendvw commented:
Excellent! Updated the tests to be much simpler, and removed some cruft that was lying around.  Would you prefer this as a flattened branch, or is this ok to merge?

bertt commented:
thanks it's in release 1.7.0
